packages/aleo-gateway/src/weighted_signers.rs

```diff
@@ -44,35 +56,104 @@ impl TryFrom<&VerifierSet> for WeightedSigners {
                     weight: Default::default(),
                 })
             }))
-            .take(32)
+            .take(GROUP_SIZE * GROUPS)
             .collect::<Result<Vec<_>, _>>()?;
 
-        signers.sort_by(|signer1, signer2| signer1.signer.cmp(&signer2.signer));
+        // signers.sort_by(|signer1, signer2| signer1.signer.cmp(&signer2.signer));
+        signers.sort_by(|signer1, signer2| {
+            /* give the lowest priority to the default address */
+            if signer1.signer == Address::default() {
+                std::cmp::Ordering::Greater
+            } else if signer2.signer == Address::default() {
+                std::cmp::Ordering::Less
+            } else {
+                signer1.signer.cmp(&signer2.signer)
+            }
+        });
```

Comment:
I've addressed this in a previous comment over the `proof.rs` file. 

I believe encapsulating this behavior, either in a function or in a wrapper type implementing `PartialOrd`, would lead to more legible code, less duplicated code.